### DISCUSSION TITLE
Welcome to oscakampala.github.io Discussions!

### DISCUSSION THREAD

**Discussion by oscakampala** [oscakampala/oscakampala.github.io#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Lincoln-developer** [comment]
Hello, team . well this is about the server side of the website, so am asking whether is a nice a nice idea to use a proxy based approach for the server-side,  where we can have our server side code within the same repo, and if it is okay then I can set up the backend folder.  Beginning, I think we may start with user account creation, sign in and login. So basically like what @KakoozaJerry said, its based on Node.js, the frameworks is Express.js and Mongo Database. So before literally start anything, Please lets open up to this discussion before any further step.  Especially for the issue of [database].

**Reply by KakoozaJerry** [reply]
I was suggesting we have a separate repository for the server side in order not to have a monolithic architecture.

So first we identify the features and create the much needed repository for then .

**Reply by Lincoln-developer** [reply]
Alright, so far with the available accomplished features mainly the frontend, there are quite alot that is done. So for the server side basically l have looked at several things
1. The sign up of users
2. Login of users
3. Events page . With this l have thought of the frontend to have an admin dashboard because and these can be the leads . Reason being we shall have to add , delete, update events, and these must be leads to play this role.
Then same with products, adding products, update or deleting them, and the ability of the user to make an order for the products.
So with the backend is to implement the products route to handle products.

**Reply by KakoozaJerry** [reply]
[OSCA WebApp project  (Aug 28, 2022).pdf](https://github.com/oscakampala/oscakampala.github.io/files/9440353/OSCA.WebApp.project.ttw-rysb-qer.-.Aug.28.2022.pdf)
